contribs/gnofaucet/captcha.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+
+	"github.com/gnolang/gno/tm2/pkg/commands"
+)
+
+type captchaCfg struct {
+	rootCfg       *serveCfg
+	captchaSecret string
+}
+
+func (c *captchaCfg) RegisterFlags(fs *flag.FlagSet) {
+	fs.StringVar(
+		&c.captchaSecret,
+		"captcha-secret",
+		"",
+		"recaptcha secret key (if empty, captcha are disabled)",
+	)
+}
+
+func newCaptchaCmd(rootCfg *serveCfg) *commands.Command {
+	cfg := &captchaCfg{
+		rootCfg: rootCfg,
+	}
+
+	return commands.NewCommand(
+		commands.Metadata{
+			Name:       "captcha",
+			ShortUsage: "captcha [flags]",
+			LongHelp:   "applies captcha middleware to the gno.land faucet",
+		},
+		cfg,
+		func(ctx context.Context, args []string) error {
+			return execCaptcha(ctx, cfg, commands.NewDefaultIO())
+		},
+	)
+}
+
+func execCaptcha(ctx context.Context, cfg *captchaCfg, io commands.IO) error {
+	if cfg.captchaSecret == "" {
+		return fmt.Errorf("captcha secret is required")
```

Comment:
This should be a standard error

Reply:
Done 82c3600f317e8a29df77f24290a24298d7332314